doc/generate_interfaces.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+
+#-----------------------------------
+# Generate documentation from icepack source code
+# Will cut and paste all code between "autodocument_start" and "autodocument_end"
+#   (defined by $startline and $endline below)
+# Writes to source/user_guide/interfaces.rst ($rstfile below)
+# Expects $startline to include a section title string such as
+#   ! autodocument_start icepack_subroutine_name
+# That string (ie. icepack_subroutine_name) will be used as the section name 
+#   in the rst documentation
+# The pattern matching assumes the title is anything following the $startline string
+#   and can include spaces
+# The file interfaces.rst is included automatically in the icepack documentation
+# To update the documentation
+#   cd to this directory
+#   run this script (./generate_interfaces.sh)
+#   git add the $rstfile
+#   git commit, push, and create PR
+# It is recommended that each section of icepack that is documented start with a
+#   comment that includes a ! in the first chararacter postition.  This will insure
+#   proper indenting of the section.
+#
+# For example:
+#   ! autodocument_start test_section
+#   ! this is an example
+#
+#      subroutine test(arg1)
+#
+#      integer, intent(in), optional :: arg1  ! test argument
+#
+#   ! autodocument_end
+#
+#-----------------------------------
+
+startline="autodocument_start"
+endline="autodocument_end"
+inpfiles="../columnphysics/*.F90"
+rstfile="./source/user_guide/interfaces.rst"
+dowrite=0
+
+mv ${rstfile} ${rstfile}.orig
+
+for file in ${inpfiles}; do
+
+filename=`basename $file`
+
+  while IFS= read -r line; do
+    if [[ $line =~ .*$endline.* ]]; then
+      dowrite=0
+      #echo "$dowrite $line"
```

Comment:
I have removed the debug lines.  I tend to leave them in because they always seem to come in handy later, but understand it's nice to leave it in a cleaner state.